libraries/AP_Compass/AP_Compass_SITL.cpp

```diff
@@ -131,15 +141,30 @@ void AP_Compass_SITL::_timer()
             // scale the first compass to simulate sensor scale factor errors
             f *= _sitl->mag_scaling;
         }
-        
-        accumulate_sample(f, _compass_instance[i], 10);
+        accumulate_sample(f, _compass_nonuc_instance[i], 10);
     }
+
+    for (uint8_t i = 0; i < MAX_CONNECTED_MAGS; i++) {
+        if (_compass_ucnode[i] != nullptr) {
+            uavcan::Publisher<uavcan::equipment::ahrs::MagneticFieldStrength2> mfs2_pub(*(_compass_ucnode[i]->get_node()));
+            uavcan::equipment::ahrs::MagneticFieldStrength2 mfs2;
+            mfs2.sensor_id = _compass_ucnode_sensor_id[i];
+            mfs2.magnetic_field_ga[0] = (double)new_mag_data.x/1000.0;
```

Comment:
why double precision?

Reply:
Don't know what got to me there, I will fix that.